src/compiler/factory.ts

```diff
@@ -3636,15 +3636,16 @@ namespace ts {
         name: "typescript:values",
         scoped: false,
         text: `
-            var __values = (this && this.__values) || function (o) {
-                var m = typeof Symbol === "function" && o[Symbol.iterator], i = 0;
+            var __values = (this && this.__values) || function(o) {
+                var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
                 if (m) return m.call(o);
-                return {
+                if (o && typeof o.length === "number") return {
                     next: function () {
                         if (o && i >= o.length) o = void 0;
                         return { value: o && o[i++], done: !o };
                     }
                 };
+                throw new TypeError(s ? "Object not iterable." : "Symbol.iterator is not defined.");

```

Comment:
Can you make this "Object **is** not iterable."?

Reply:
Adding a suggestion.

```suggestion
                throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
```

Reply:
There are too many tests to resolve doing these via suggestions...